### PR TITLE
[SYNCOPE-1658] Tabular Topology

### PR DESCRIPTION
**Pull request by SamuelGaro**

### CODE REVIEW

File: client/console/src/main/java/org/apache/syncope/client/console/commons/ConnectorDataProvider.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+        return comparator;
+    }
+
+    @Override
+    public long size() {
+        long result = 0;
+
+        try {
+            if (keyword == null || keyword.isEmpty()) {
```
Comment (ilgrosso) [inline client/console/src/main/java/org/apache/syncope/client/console/commons/ConnectorDataProvider.java:126]: Use `StringUtils#isBlank` instead

File: client/console/src/main/java/org/apache/syncope/client/console/commons/ResourceDataProvider.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    @Override
+    public Iterator<ResourceTO> iterator(final long first, final long count) {
+        List<ResourceTO> result = Collections.emptyList();
+
+        try {
+            currentPage = ((int) first / paginatorRows);
+            if (currentPage < 0) {
+                currentPage = 0;
+            }
+            if (keyword == null || keyword.isEmpty()) {
```
Comment (ilgrosso) [inline client/console/src/main/java/org/apache/syncope/client/console/commons/ResourceDataProvider.java:75]: Use `StringUtils#isBlank` instead

File: client/console/src/main/java/org/apache/syncope/client/console/commons/ResourceDataProvider.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+        return comparator;
+    }
+
+    @Override
+    public long size() {
+        long result = 0;
+
+        try {
+            if (keyword == null || keyword.isEmpty()) {
```
Comment (ilgrosso) [inline client/console/src/main/java/org/apache/syncope/client/console/commons/ResourceDataProvider.java:126]: Use `StringUtils#isBlank` instead

File: client/console/src/main/java/org/apache/syncope/client/console/panels/ConnectorDirectoryPanel.java [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+
+        restClient = builder.restClient;
+    }
+
+    @Override
+    public void onEvent(final IEvent<?> event) {
+        if (event.getPayload() instanceof ConnectorSearchEvent) {
+            ConnectorSearchEvent payload = (ConnectorSearchEvent) event.getPayload();
+            AjaxRequestTarget target = payload.getTarget();
+            if (payload.getKeyword() != null && !payload.getKeyword().isEmpty()) {
```
Comment (ilgrosso) [inline client/console/src/main/java/org/apache/syncope/client/console/panels/ConnectorDirectoryPanel.java:85]: Use `StringUtils#isNotBlank` instead

**Review by ilgrosso — CHANGES_REQUESTED** [review]

**Review by ilgrosso — APPROVED** [review]

**Comment by ilgrosso** [comment]
Thanks for your contribution @SamuelGaro 

Since this appears to be your very first one, please go ahead and [submit your ICLA](https://syncope.apache.org/contributing#How_do_I_become_a_contributor_or_a_committer.3F)

**Comment by SamuelGaro** [comment]
> Thanks for your contribution @SamuelGaro
> 
> Since this appears to be your very first one, please go ahead and [submit your ICLA](https://syncope.apache.org/contributing#How_do_I_become_a_contributor_or_a_committer.3F)

I sent the ICLA few minutes ago

**Comment by ilgrosso** [comment]
Thank you again @SamuelGaro !

**Pull request merged**